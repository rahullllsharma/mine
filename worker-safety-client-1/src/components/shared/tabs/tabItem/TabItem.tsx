import type { PropsWithClassName } from "@/types/Generic";
import type { IconName } from "@urbint/silica";
import cx from "classnames";
import { Icon } from "@urbint/silica";

type TabItemProps = PropsWithClassName<{
  value: string;
  icon?: IconName;
  type?: "regular" | "menu";
}>;

export default function TabItem({
  value,
  icon,
  type = "regular",
  className,
}: TabItemProps): JSX.Element {
  return (
    <div
      className={cx(
        "flex items-center",
        { ["px-2.5 py-3"]: type === "regular" },
        { ["flex-col h-full gap-0 pt-2"]: type === "menu" },
        className
      )}
    >
      {icon && (
        <Icon
          name={icon}
          className={cx({
            ["text-xl mr-1 text-neutral-shade-75"]: type === "regular",
            ["text-2xl w-7 h-7 leading-7"]: type === "menu",
          })}
        />
      )}
      <span
        className={cx({
          ["text-base mx-1 truncate"]: type === "regular",
          ["text-sm truncate"]: type === "menu",
        })}
      >
        {value}
      </span>
    </div>
  );
}
