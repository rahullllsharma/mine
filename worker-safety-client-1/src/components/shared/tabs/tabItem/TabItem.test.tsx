import { render, screen } from "@testing-library/react";
import TabItem from "./TabItem";

describe(TabItem.name, () => {
  it('should render with a text element with the same value given by the "value" prop', () => {
    render(<TabItem value="Lorem ipsum" />);
    screen.getByText(/lorem ipsum/i);
  });

  it('should render with an icon if the "icon" prop is passed', () => {
    const { asFragment } = render(
      <TabItem value="Lorem ipsum" icon="settings_future" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
