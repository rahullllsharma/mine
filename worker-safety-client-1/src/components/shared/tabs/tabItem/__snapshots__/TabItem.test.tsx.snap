// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem should render with an icon if the "icon" prop is passed 1`] = `
<DocumentFragment>
  <div
    class="flex items-center px-2.5 py-3"
  >
    <i
      aria-hidden="true"
      class="ci-settings_future text-xl mr-1 text-neutral-shade-75"
    />
    <span
      class="text-base mx-1 truncate"
    >
      Lorem ipsum
    </span>
  </div>
</DocumentFragment>
`;
